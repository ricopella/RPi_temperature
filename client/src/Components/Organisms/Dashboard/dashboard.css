.dashboard-container {
    height: 100%;
}

.dashboard-header {
    display: grid;
    text-align: center;
}

.dashboard-body {
    margin-top: 20px;
    display: grid;
    text-align: center;
}

.dashboard-body i {
    font-size: 2em;
    display: grid;
    align-items: center;
}

button {
    color: #ffffff;
    background-color: #006400;
    font-size: 1.5em;
    width: 400px;
    height: 60px;
    border: none;
    margin: 30px 18% 0 18%;
}