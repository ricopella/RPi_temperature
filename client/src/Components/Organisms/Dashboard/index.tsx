import { AxiosResponse } from 'axios';
import * as React from 'react';

import { CurrentTempResponse } from '../../../types';
import { formatDate, removeLineBreak } from '../../../utils/strings';
import Tile from '../../Atoms/Tile';

import './dashboard.css';

export interface DashboardProps {
  getTemp: () => void;
  mostRecent: CurrentTempResponse | null;
  temperature: AxiosResponse<any> | null;
}

export default class Dashboard extends React.PureComponent<
  DashboardProps,
  any
> {
  public render() {
    const { getTemp, mostRecent, temperature } = this.props;

    const date =
      mostRecent && mostRecent.createdAt
        ? formatDate(mostRecent.createdAt)
        : 'N/A';
    return (
      <div className="dashboard-container">
        <Tile>
          <div className="dashboard-header">
            <h3>
              Most Recent Temperature:{' '}
              {mostRecent && mostRecent.temperature
                ? `${removeLineBreak(mostRecent.temperature)}.`
                : 'N/A'}
            </h3>
            <span>
              <i>Last checked: {date}</i>
            </span>
          </div>
          <div className="dashboard-body">
            Current Temperature is: <i>{temperature ? temperature : 'N/A'}</i>
          </div>
          <button onClick={getTemp}>Get Temperature!</button>
        </Tile>
      </div>
    );
  }
}
