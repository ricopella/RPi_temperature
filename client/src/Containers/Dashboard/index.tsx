import { AxiosResponse } from 'axios';
import * as React from 'react';

import Dashboard from '../../Components/Organisms/Dashboard';
import API from '../../Helpers';
import { CurrentTempResponse } from '../../types';

export interface DashboardState {
  mostRecent: CurrentTempResponse | null;
  temp: AxiosResponse<any> | null;
}

export default class DashboardContainer extends React.Component<
  {},
  DashboardState
> {
  constructor(props: {}) {
    super(props);

    this.state = {
      mostRecent: null,
      temp: null,
    };
  }

  public componentDidMount() {
    API.getMostRecent().then(res => {
      this.setState(
        {
          mostRecent: res.data.temp[0],
        },
        () => console.log(this.state.mostRecent),
      );
    });
  }

  private getTemp = () => {
    API.getCurrentTemp()
      .then(res => {
        console.log({ res });
        this.setState({
          temp: res.data.temp,
        });
      })
      .catch(err => {
        console.log('ERROR Container: ', { err });
      });
  };

  public render() {
    const { mostRecent, temp } = this.state;
    return (
      <React.Fragment>
        <Dashboard
          getTemp={this.getTemp}
          temperature={temp}
          mostRecent={mostRecent}
        />
      </React.Fragment>
    );
  }
}
