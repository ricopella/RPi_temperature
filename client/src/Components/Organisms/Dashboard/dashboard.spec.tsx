import { mount } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Dashboard from '.';

const props = {
  getTemp: () => null,
  mostRecent: null,
  temperature: null,
};

describe('<Dashboard />', () => {
  it('should match internal tree', () => {
    const internalTree = renderer.create(<Dashboard {...props} />).toJSON();

    expect(internalTree).toMatchSnapshot();
  });

  it('should call getTemp onClick', () => {
    const spy = jest.fn();
    props.getTemp = spy;

    const component = mount(<Dashboard {...props} />);
    const button = component.find('button');

    button.simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it('should show N/A when no most recent response', () => {
    const component = mount(<Dashboard {...props} />);
    const actual = component.find('h3').text();
    const expected = 'Most Recent Temperature: N/A';

    expect(actual).toEqual(expected);
  });

  it('should show N/A when no temperature', () => {
    const component = mount(<Dashboard {...props} />);
    const actual = component.find('.dashboard-body').text();
    const expected = 'Current Temperature is: N/A';

    expect(actual).toEqual(expected);
  });

  it('should show most recent temperature when no most recent response', () => {
    // @ts-ignore
    props.mostRecent = {
      __V: 0,
      _id: '0',
      createdAt: '2018-08-12T07:00:00.000Z',
      temperature: '80\n',
      updatedAt: '2018-08-12T07:00:00.000Z',
    };
    const component = mount(<Dashboard {...props} />);
    const actual = component.find('h3').text();
    const expected = 'Most Recent Temperature: 80.';

    expect(actual).toEqual(expected);
  });

  it('should show most recent temperature when no most recent response', () => {
    // @ts-ignore
    props.temperature = '83';
    const component = mount(<Dashboard {...props} />);
    const actual = component.find('.dashboard-body').text();
    const expected = 'Current Temperature is: 83';

    expect(actual).toEqual(expected);
  });
});
