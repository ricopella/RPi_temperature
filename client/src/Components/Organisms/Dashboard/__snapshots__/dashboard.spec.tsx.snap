// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match internal tree 1`] = `
<div
  className="dashboard-container"
>
  <div
    className="tile"
  >
    <div
      className="dashboard-header"
    >
      <h3>
        Most Recent Temperature: 
        N/A
      </h3>
      <span>
        <i>
          Last checked: 
          N/A
        </i>
      </span>
    </div>
    <div
      className="dashboard-body"
    >
      Current Temperature is: 
      <i>
        N/A
      </i>
    </div>
    <button
      onClick={[Function]}
    >
      Get Temperature!
    </button>
  </div>
</div>
`;
